app/services/moderator/merge_user.rb

```diff
@@ -55,45 +58,77 @@ def update_social
       end
     end
 
+    def merge_sponsorships
+      @delete_user.sponsorships.update_all(user_id: @keep_user.id)
+    end
+
     def merge_profile
-      if @delete_user.github_repos.any?
-        @delete_user.github_repos.update_all(user_id: @keep_user.id)
-        @keep_user.touch(:github_repos_updated_at)
-      end
-      if @delete_user.badge_achievements.any?
-        @delete_user.badge_achievements.update_all(user_id: @keep_user.id)
-        BadgeAchievement.counter_culture_fix_counts(where: { users: { id: @keep_user.id } })
-      end
+      @delete_user.github_repos.update_all(user_id: @keep_user.id)
+      @keep_user.touch(:github_repos_updated_at)
 
       @keep_user.update_columns(created_at: @delete_user.created_at) if @delete_user.created_at < @keep_user.created_at
     end
 
-    def merge_chat_mentions
-      any_memberships = @delete_user.chat_channel_memberships.any?
-      @delete_user.chat_channel_memberships.update_all(user_id: @keep_user.id) if any_memberships
-      @delete_user.mentions.update_all(user_id: @keep_user.id) if @delete_user.mentions.any?
+    def merge_badge_achievements
+      duplicate_badges = @delete_user.badge_achievements.pluck(:badge_id).intersection(
+        @keep_user.badge_achievements.pluck(:badge_id),
+      )
+      @delete_user.badge_achievements.where(badge_id: duplicate_badges).delete_all
+      @delete_user.badge_achievements.update_all(user_id: @keep_user.id)
+      BadgeAchievement.counter_culture_fix_counts(where: { users: { id: @keep_user.id } })
     end
 
-    def merge_follows
-      @delete_user.follows&.update_all(follower_id: @keep_user.id) if @delete_user.follows.any?
-      @delete_user_followers = Follow.followable_user(@delete_user.id)
-      @delete_user_followers.update_all(followable_id: @keep_user.id) if @delete_user_followers.any?
+    def merge_chat_channels
+      duplicate_moderator_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active", role: "mod",
+      ).pluck(:chat_channel_id).intersection(
+        @keep_user.chat_channel_memberships.where(
+          status: "active", role: "mod",
+        ).pluck(:chat_channel_id),
+      )
+      @delete_user.chat_channel_memberships.where(chat_channel_id: duplicate_moderator_memberships).delete_all
+
+      moderator_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active", role: "mod",
+      ).pluck(:chat_channel_id)
+
+      duplicate_memberships = @delete_user.chat_channel_memberships.where(
+        status: "active",
+      ).pluck(:chat_channel_id).intersection(
```

Comment:
It looks a little odd to do an intersection on two arrays plucked from the database, rather than a join query, to get this list. 

I think we can use a pattern like this (passing a selected column rather than an array of ids) to get the overlap from the db as a relation, rather than as an array of ids?

```
@delete_user
  .chat_channel_memberships
  .where(
    status: :active, 
    chat_channel_id: 
      @keep_user.chat_channel_memberships.where(status: :active).select(:chat_channel_id)
    )

  ChatChannelMembership Load (2.3ms)  SELECT "chat_channel_memberships".* FROM "chat_channel_memberships" WHERE "chat_channel_memberships"."user_id" = $1 AND "chat_channel_memberships"."status" = $2 AND "chat_channel_memberships"."chat_channel_id" IN (SELECT "chat_channel_memberships"."chat_channel_id" FROM "chat_channel_memberships" WHERE "chat_channel_memberships"."user_id" = $3 AND "chat_channel_memberships"."status" = $4)  [["user_id", 4], ["status", "active"], ["user_id", 2], ["status", "active"]]

=> [#<ChatChannelMembership:0x000055b12e09b2a0
  id: 7,
  chat_channel_id: 3,
  created_at: Tue, 23 Mar 2021 16:08:44 UTC +00:00,
  has_unopened_messages: false,
  last_opened_at: Sun, 01 Jan 2017 05:00:00 UTC +00:00,
  role: "mod",
  show_global_badge_notification: true,
  status: "active",
  updated_at: Tue, 23 Mar 2021 16:08:44 UTC +00:00,
  user_id: 4>]
```

This has the secondary advantage of returning a relation (that responds to `#delete_all`). 

I think it's a little wordy to type, but the generated query looks correct (and this pattern can be applied elsewhere).